slack-lint/src/main/java/slack/lint/text/SpanPointMarkDangerousCheckDetector.kt

```diff
@@ -0,0 +1,122 @@
+import org.jetbrains.uast.UastBinaryOperator
+import org.jetbrains.uast.tryResolve
+import slack.lint.text.SpanPointMarkDangerousCheckDetector.Companion.ISSUE
+import slack.lint.util.sourceImplementation
+
+/**
+ * Checks for SpanPointMarkDangerousCheck. See [ISSUE].
+ */
+class SpanPointMarkDangerousCheckDetector : Detector(), SourceCodeScanner {
+
+  companion object {
+    private fun Implementation.toIssue() = Issue.create(
+      id = "SpanPointMarkDangerousCheck",
+      briefDescription = "Check Span Flags Using Bitmask",
+      explanation =
+      "Spans flags can have priority or other bits set. " +
+        "Ensure that Span flags are checked using " +
+        "`currentFlag and Spanned.SPAN_POINT_MARK_MASK == desiredFlag` " +
+        "rather than just `currentFlag == desiredFlag`",
+      category = Category.CORRECTNESS,
+      priority = 4,
+      severity = Severity.ERROR,
+      implementation = this
+    )
+
+    val ISSUE = sourceImplementation<SpanPointMarkDangerousCheckDetector>().toIssue()
+  }
+
+  override fun getApplicableUastTypes() = listOf(UBinaryExpression::class.java)
+
+  override fun createUastHandler(context: JavaContext): UElementHandler {
+    return ReportingHandler(context)
+  }
+
+  /**
+   * Reports violations of SpanPointMarkDangerousCheck.
+   */
+  private class ReportingHandler(private val context: JavaContext) : UElementHandler() {
+    companion object {
+      private const val SPANNED_CLASS = "android.text.Spanned"
+      private val markPointFields = setOf(
+        "$SPANNED_CLASS.SPAN_INCLUSIVE_INCLUSIVE",
+        "$SPANNED_CLASS.SPAN_INCLUSIVE_EXCLUSIVE",
+        "$SPANNED_CLASS.SPAN_EXCLUSIVE_INCLUSIVE",
+        "$SPANNED_CLASS.SPAN_EXCLUSIVE_EXCLUSIVE",
+      )
+      private const val MASK_CLASS = "$SPANNED_CLASS.SPAN_POINT_MARK_MASK"
+    }
+
+    override fun visitBinaryExpression(node: UBinaryExpression) {
+      if (node.operator is UastBinaryOperator.ComparisonOperator) {
+        checkExpressions(node, node.leftOperand, node.rightOperand)
+        checkExpressions(node, node.rightOperand, node.leftOperand)
+      }
+    }
+
+    fun checkExpressions(node: UBinaryExpression, markPointCheck: UExpression, maskCheck: UExpression) {
+      if (matchesMarkPoint(markPointCheck) && !matchesMask(maskCheck)) {
+        context.report(
+          ISSUE,
+          context.getLocation(node),
+          "Do not check against ${markPointCheck.sourcePsi?.text} directly. " +
+            "Instead mask flag with Spanned.SPAN_POINT_MARK_MASK to only check MARK_POINT flags.",
+          LintFix.create()
+            .replace()
+            .name("Use bitwise mask")
+            .text(maskCheck.sourcePsi?.text)
+            .with("((${maskCheck.sourcePsi?.text}) and $MASK_CLASS)")
+            .build()
+        )
+      }
+    }
+
+    fun matchesMarkPoint(expression: UExpression): Boolean {
+      return markPointFields.contains(getQualifiedName(expression))
```

Comment:
nit: use the `in` operator?

Reply:
👍 fixed in https://github.com/slackhq/slack-lints/pull/14/commits/cf6ddb06a9ac8db063ed9c045a2bad955396c5a7